From 9ffa4f916a5639ae60f9de52902a4587d492816d Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Mon, 11 Sep 2017 10:46:52 +0800
Subject: [PATCH] rtc: class: rockchip: print time spent in suspend

Log likes:
[   28.043238] Suspending console(s) (use no_console_suspend to debug)
INFO:    sleep mode config[0xfe]:
...
[   28.246039] PM: suspend of devices complete after 201.289 msecs
[   28.260911] PM: late suspend of devices complete after 3.238 msecs
[   28.262877] PM: noirq suspend of devices complete after 1.949 msecs
[   28.262885] Disabling non-boot CPUs ...
...
[   28.494444] Enabling non-boot CPUs ...
...
[   28.603791] PM: noirq resume of devices complete after 1.982 msecs
[   28.606132] PM: early resume of devices complete after 1.713 msecs
...
[   28.696963] Suspended for 2.265 seconds

Change-Id: If6887e62c5319f7fd39455db54190c84e5eb366b
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/rtc/class.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/rtc/class.c b/drivers/rtc/class.c
index de86578bcd6d..c795ef547357 100644
--- a/drivers/rtc/class.c
+++ b/drivers/rtc/class.c
@@ -141,6 +141,14 @@ static int rtc_resume(struct device *dev)
 	if (sleep_time.tv_sec >= 0)
 		timekeeping_inject_sleeptime64(&sleep_time);
 	rtc_hctosys_ret = 0;
+
+#ifdef CONFIG_ARCH_ROCKCHIP
+	if (sleep_time.tv_sec >= 0)
+		printk("Suspended for %lu.%03lu seconds\n",
+			(unsigned long)sleep_time.tv_sec,
+			sleep_time.tv_nsec / NSEC_PER_MSEC);
+#endif
+
 	return 0;
 }
 
-- 
2.35.3

